environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -116,23 +203,120 @@ public Weld(String containerId) {
      *
      * @param extension an extension
      */
-    public void addExtension(Extension extension) {
+    public Weld addExtension(Extension extension) {
         if (extensions == null) {
             extensions = new HashSet<Metadata<Extension>>();
         }
         extensions.add(new MetadataImpl<Extension>(extension, "<explicitly-added>"));
+        return this;
+    }
+
+    /**
+     * Enable interceptors for a synthetic bean archive. Interceptor classes are automatically added to the set of bean classes for the synthetic bean archive.
+     *
+     * @param interceptorClasses
+     * @return self
+     */
+    public Weld interceptors(Class<?>... interceptorClasses) {
+        enabledInterceptors.clear();
+        for (Class<?> interceptorClass : interceptorClasses) {
+            beanClasses.add(interceptorClass.getName());
+            enabledInterceptors.add(syntheticMetadata(interceptorClass));
+        }
+        return this;
+    }
+
+    /**
+     * Enable decorators for a synthetic bean archive. Decorator classes are automatically added to the set of bean classes for the synthetic bean archive.
+     *
+     * @param decoratorClasses
+     * @return self
+     */
+    public Weld decorators(Class<?>... decoratorClasses) {
+        enabledDecorators.clear();
+        for (Class<?> decoratorClass : decoratorClasses) {
+            beanClasses.add(decoratorClass.getName());
+            enabledDecorators.add(syntheticMetadata(decoratorClass));
+        }
+        return this;
+    }
+
+    /**
+     * Select alternatives for a synthetic bean archive.
+     *
+     * @param alternativeClasses
+     * @return self
+     */
+    public Weld alternatives(Class<?>... alternativeClasses) {
+        selectedAlternatives.clear();
+        for (Class<?> alternativeClass : alternativeClasses) {
+            selectedAlternatives.add(syntheticMetadata(alternativeClass));
+        }
+        return this;
     }
 
     /**
-     * Boots Weld and creates and returns a WeldContainer instance, through which
-     * beans and events can be accessed.
+     * Select alternative stereotypes for a synthetic bean archive.
      *
-     * @return weld container
+     * @param alternativeStereotypeClasses
+     * @return self
+     */
+    public Weld alternativeStereotypes(Class<?>... alternativeStereotypeClasses) {
+        selectedAlternativeStereotypes.clear();
+        for (Class<?> alternativeStereotypClass : alternativeStereotypeClasses) {
+            selectedAlternativeStereotypes.add(syntheticMetadata(alternativeStereotypClass));
+        }
+        return this;
+    }
+
+    /**
+     * Reset the enablement for a synthetic bean archive.
+     *
+     * @return self
+     */
+    public Weld resetEnablement() {
```

Comment:
Yes, that would make more sense. The original idea was based on the needs of the test :-)